lms/djangoapps/support/static/support/js/views/certificates.js

```diff
@@ -0,0 +1,151 @@
+                // This is useful because it allows users to share search results:
+                // if the URL contains ?query="foo" then anyone who loads that URL
+                // will automatically search for "foo".
+                if (this.initialQuery) {
+                    this.setUserQuery(this.initialQuery);
+                    this.triggerSearch();
+                }
+
+                return this;
+            },
+
+            renderResults: function() {
+                var context = {
+                    certificates: this.certificates,
+                };
+
+                this.setResults(_.template(resultsTpl, context));
+            },
+
+            renderError: function(error) {
+                var errorMsg = error || gettext("An unexpected error occurred.  Please try again.");
+                this.setResults(errorMsg);
+            },
+
+            search: function(event) {
+
+                // Fetch the certificate collection for the given user
+                var query = this.getUserQuery(),
+                    url = "/support/certificates?query=" + query;
+
+                // Prevent form submission, since we're handling it ourselves.
+                event.preventDefault();
+
+                // Push a URL into history with the search query as a GET parameter.
+                // That way, if the user reloads the page or sends someone the link
+                // then the same search will be performed on page load.
+                window.history.pushState({}, window.document.title, url);
+
+                // Perform a search for the user's certificates.
+                this.disableButtons();
+                this.certificates.setUserQuery(query);
+                this.certificates.fetch({
+                    success: this.updateCertificates,
+                    error: this.handleSearchError
+                });
+            },
+
+            regenerateCertificate: function(event) {
+                var $button = $(event.target);
+
+                // Regenerate certificates for a particular user and course.
+                // If this is successful, reload the certificate results so they show
+                // the updated status.
+                this.disableButtons();
+                $.ajax({
+                    url: "/certificates/regenerate",
+                    type: "POST",
+                    data: {
+                        username: $button.data("username"),
+                        course_key: $button.data("course-key"),
+                    },
+                    context: this,
+                    success: function() {
+                        this.certificates.fetch({
+                            success: this.updateCertificates,
+                            error: this.handleSearchError,
+                        });
+                    },
+                    error: this.handleRegenerateError
+                });
```

Comment:
I'd be surprised if we wanted to initiate this AJAX request from other views.  If it's okay, I'd prefer to keep this as an AJAX call, since I think making it a model will add complexity (separate model file, setting up listeners for events) without much payoff.  If we need to factor this out later, we can always do so.